### PR TITLE
Add task CoffeeShop

### PR DESCRIPTION
**Pull request by VolhaVaitulevich**
Link: https://edabit.com/challenge/ydgurBgEdGkueNmmp

### CODE REVIEW

File: exercises/classes/coffeeShop/index.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		return fulfillResult;
+	}
+
+	listOrders(): Array<String> {
+		return this.orders;
+	}
+
+	dueAmount(): number {
+		const dueOrderAmount = this.orders.reduce((totalOrderPrice, orderItem) => {
+			return (totalOrderPrice +=
```
Comment (siarhei-papkou) [inline exercises/classes/coffeeShop/index.ts]: +=

File: exercises/classes/coffeeShop/index.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	}
+
+	drinksOnly(): String[] {
+		const listOfDrinks = this.menu
+			.filter(menuObject => menuObject.type === 'drink')
+			.map(menuObject => menuObject.item);
+		return listOfDrinks;
+	}
+
+	foodOnly(): String[] {
```
Comment (siarhei-papkou) [inline exercises/classes/coffeeShop/index.ts]: string

File: exercises/classes/coffeeShop/index.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		return dueOrderAmount;
+	}
+
+	cheapestItem(): string {
+		const minPrice = this.menu.reduce(
+			(min, menuObject) => (menuObject.price < min ? menuObject.price : min),
+			this.menu[0].price,
+		);
+		const minPriceName: string =
+			this.menu.find(menuObject => menuObject.price === minPrice)?.item ?? '';
```
Comment (siarhei-papkou) [inline exercises/classes/coffeeShop/index.ts]: .

File: exercises/classes/coffeeShop/index.ts (hunk 1 above)
Comment (VolhaVaitulevich) [reply exercises/classes/coffeeShop/index.ts]: Updated in a new commit

File: exercises/classes/coffeeShop/index.ts (hunk 3 above)
Comment (VolhaVaitulevich) [reply exercises/classes/coffeeShop/index.ts]: Updated in a new commit

**Review by siarhei-papkou — APPROVED** [review]

**Pull request merged**